### PR TITLE
Fix for compiling on MINGW/MSYS2 and possibly Cygwin

### PR DESCRIPTION
**Pull request by emmenlau**
Fixes issue #89 and enables successful compile on MSYS2. Possibly helps the compile on MINGW and Cygwin, too.

### CODE REVIEW

File: javabridge/locate.py [hunk 1]
```diff
@@ -15,14 +15,39 @@
...
+
+# Note: if a matching gcc is available from the shell on Windows, its
+#       probably safe to assume the user is in an MINGW or MSYS or Cygwin
+#       environment, in which case he/she wants to compile with gcc for
+#       Windows, in which case the correct compiler flags will be triggered
+#       by is_mingw. This method is not great, improve it if you know a
+#       better way to discriminate between compilers on Windows.
+def is_mingw():
+    if (os.name=="nt"):
+        mingw32 = ""
```
Comment (LeeKamentsky) [inline javabridge/locate.py]: I would put a check in here to see if the environment variables, MSSdk and DISTUTILS_USE_SDK are set to values other than zero (see https://github.com/cython/cython/wiki/CythonExtensionsOnWindows). That's an indication of the  user's intent - to compile using the Microsoft toolchain.

**Comment by LeeKamentsky** [comment]
Thanks, Mario. I'm just getting back from vacation. I'd like to test the build on a Windows machine with the Microsoft Visual C++ compiler for Python 2.7 (https://www.microsoft.com/en-us/download/details.aspx?id=44266) before accepting the PR. If you've tested it, that's good enough for me.

File: javabridge/locate.py (hunk 1 above)
Comment (emmenlau) [inline javabridge/locate.py]: I've just updated the PR to include such a change. Now it should be rather conservative on guessing mingw.

**Comment by emmenlau** [comment]
Welcome back! :-)
I haven't tested with anything other than MSYS2, so indeed it might be good to make a test. Do you have access to a Microsoft Visual C++ test environment, or know someone who could test for us?

**Pull request merged**